cmd/flux/bootstrap_stash.go

```diff
@@ -0,0 +1,261 @@
+
+  # Run bootstrap for a private repository using SSH authentication
+  flux bootstrap stash --owner=<project> --username=<user> --repository=<repository name> --hostname=<domain>
+
+  # Run bootstrap for a repository path
+  flux bootstrap stash --owner=<project> --username=<user> --repository=<repository name> --path=dev-cluster --hostname=<domain>
+
+  # Run bootstrap for a public repository on a personal account
+  flux bootstrap stash --owner=<user> --repository=<repository name> --private=false --personal --hostname=<domain> --token-auth
+
+  # Run bootstrap for a an existing repository with a branch named main
+  flux bootstrap stash --owner=<project> --username=<user> --repository=<repository name> --branch=main --hostname=<domain> --token-auth`,
+	RunE: bootstrapStashCmdRun,
+}
+
+const (
+	stashDefaultPermission = "push"
+	stashTokenEnvVar       = "STASH_TOKEN"
+)
+
+type stashFlags struct {
+	owner        string
+	repository   string
+	interval     time.Duration
+	personal     bool
+	username     string
+	private      bool
+	hostname     string
+	path         flags.SafeRelativePath
+	teams        []string
+	readWriteKey bool
+	reconcile    bool
+}
+
+var stashArgs stashFlags
+
+func init() {
+	bootstrapStashCmd.Flags().StringVar(&stashArgs.owner, "owner", "", "Stash user or project name")
+	bootstrapStashCmd.Flags().StringVar(&stashArgs.repository, "repository", "", "Stash repository name")
+	bootstrapStashCmd.Flags().StringSliceVar(&stashArgs.teams, "group", []string{}, "Stash groups to be given write access (also accepts comma-separated values)")
+	bootstrapStashCmd.Flags().BoolVar(&stashArgs.personal, "personal", false, "if true, the owner is assumed to be a Stash user; otherwise a group")
+	bootstrapStashCmd.Flags().StringVarP(&stashArgs.username, "username", "u", "git", "authentication username")
+	bootstrapStashCmd.Flags().BoolVar(&stashArgs.private, "private", true, "if true, the repository is setup or configured as private")
+	bootstrapStashCmd.Flags().DurationVar(&stashArgs.interval, "interval", time.Minute, "sync interval")
+	bootstrapStashCmd.Flags().StringVar(&stashArgs.hostname, "hostname", "", "Stash hostname")
+	bootstrapStashCmd.Flags().Var(&stashArgs.path, "path", "path relative to the repository root, when specified the cluster sync will be scoped to this path")
+	bootstrapStashCmd.Flags().BoolVar(&stashArgs.readWriteKey, "read-write-key", false, "if true, the deploy key is configured with read/write permissions")
+	bootstrapStashCmd.Flags().BoolVar(&stashArgs.reconcile, "reconcile", false, "if true, the configured options are also reconciled if the repository already exists")
+
+	bootstrapCmd.AddCommand(bootstrapStashCmd)
+}
+
+func bootstrapStashCmdRun(cmd *cobra.Command, args []string) error {
+	stashToken := os.Getenv(stashTokenEnvVar)
+	if stashToken == "" {
+		return fmt.Errorf("%s environment variable not found", stashTokenEnvVar)
+	}
```

Comment:
Do not think this does need addressing in this PR, but I do think that we do this check (with a generic error) in a couple more places now, so it may be easier to factor it out into something that returns `(string, error)`.

For example (with improvised func name):

```go
func MustGetenv(var string) (string, error) {
	v := os.Getenv(var)
	if v == "" {
		return "", fmt.Errorf("%s environment variable not found or empty", var)
	}
        return v, nil
}
```

Reply:
We no longer require the tokens to be setup as env vars, see #2038

Reply:
updated with the `prompt`code.